src/main/java/org/edumips64/Main.java

```diff
@@ -55,10 +55,10 @@
 
 public class Main extends JApplet {
 
-  private static String VERSION;
-  private static String CODENAME;
-  private static String BUILD_DATE;
-  private static String GIT_REVISION;
+  public static String VERSION;
```

Comment:
Oh ok I see, you use them in CPUSwingWorker.

Would it be possible to pass them to CPUSwingWorker at construction time or with setters? We had the pattern of getting things from `Main` all across the simulator and we spent some time trying to undo it, since it created a giant global dependency.

If it's too much work it's ok to leave it as it is.

Reply:
I think that ideally that info (version etc) should live in a static singleton accessible from everywhere, and not be passed around across functions and classes. That said, Main is not the best place anyway, because for instance those info should be used also in MainCLI.  
Feel free to suggest what's the best location (create a new singleton class, or put them in MetaInfo), or push to this branch.

Reply:
I think `MetaInfo` could be extended to serve this purpose as well, thanks!